core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -565,7 +565,7 @@ public PlanNode visitLimit(LimitNode node, RewriteContext<Set<Symbol>> context)
                     .addAll(context.get())
                     .addAll(node.getTiesResolvingScheme().map(OrderingScheme::getOrderBy).orElse(ImmutableList.of()));
             PlanNode source = context.rewrite(node.getSource(), expectedInputs.build());
-            return new LimitNode(node.getId(), source, node.getCount(), node.getTiesResolvingScheme(), node.isPartial());
+            return new LimitNode(node.getId(), source, node.getCount(), node.getTiesResolvingScheme(), node.isPartial(), node.getInputOrdering());
```

Comment:
This will prune any columns in the source node that are referenced in the ordering scheme. This method needs to account for that.

Reply:
@raunaqmorarka the rule `PruneLimitColumns` needs a similar fix.

Reply:
Fixed that now, thanks